src/connections/motherduck.ts

```diff
@@ -0,0 +1,183 @@
+import { QueryType } from '../query-params'
+import { Query } from '../query'
+import { Connection } from './index'
+import {
+    Database,
+    Schema,
+    Table,
+    TableColumn,
+    TableIndex,
+    TableIndexType,
+} from '../models/database'
+import { DefaultDialect } from '../query-builder/dialects/default'
+import duckDB, { Callback, DuckDbError } from 'duckdb'
+
+const runQuery = async (
+    connection: duckDB.Connection,
+    query: string,
+    ...params: any[]
+) => {
+    return new Promise((resolve, reject) => {
+        console.log(query, ...params)
+        connection.all(query, ...params, (err, res) => {
+            if (err) {
+                reject(err)
+            } else {
+                resolve(res)
+            }
+        })
+    })
+}
+
+export class DuckDBConnection implements Connection {
+    queryType = QueryType.positional
+
+    dialect = new DefaultDialect()
+    duckDB: duckDB.Database | undefined
+
+    constructor(
+        path: string,
```

Comment:
Currently in other connections we have the constructor just take an object instead of multiple parameters. This allows any connection to have a predictable pattern. I'd say we should probably lean in that direction here as well unless we have a good reason not to and then I'd remain open-minded.